tsfm_public/toolkit/gluonts_data_wrapper.py

```diff
@@ -0,0 +1,245 @@
+import bisect
+from typing import Union
+
+import numpy as np
+import torch
+from gluonts.dataset.split import InputDataset, LabelDataset, TrainingDataset
+from gluonts.itertools import batcher
+from gluonts.transform.feature import LastValueImputation
+from torch.utils.data import Dataset
+from tqdm import tqdm
+
+
+def impute_series(target):
+    if np.isnan(target).any():
+        target = target.copy()
+        if len(target.shape) == 2:
+            for i in range(target.shape[0]):
+                target[i, ...] = LastValueImputation()(target[i, ...])
+        elif len(target.shape) == 1:
+            target = LastValueImputation()(target)
+        else:
+            raise Exception("Only 1D and 2D arrays are accepted by the impute_series() function.")
+    return target
+
+
+def np_to_torch(np):
+    if np.dtype == "float" or np.dtype == "float32":
+        return torch.from_numpy(np).float()
+    elif np.dtype == "int":
+        return torch.from_numpy(np)
+
+
+def _torch(*nps):
```

Comment:
Similar function may exist in `dataset.py`

Reply:
Thanks. https://github.com/ibm-granite/granite-tsfm/pull/219/commits/3598090ba2dca854d9af8226e1b9e814b3471026